tests/test_utilities.py

```diff
@@ -156,6 +156,159 @@ def test_check_all_contiguous():
+    g = galsim.Gaussian(sigma=3.7,flux=1000.*n*n)
+    gal = galsim.Convolve([g,galsim.Pixel(1.0)])
+    gal.drawImage(image=im,method='no_pixel',offset=galsim.PositionD(0.0,0.0),scale=1.0/n)
+    np.testing.assert_array_equal(img.array,im.array,\
+        err_msg="Interleaved Gaussian images do not match")
+    assert im.scale == img.scale
+
+    # 1b) With im_list and offsets permuted
+    offset_list = []
+    # An elegant way of generating the default offsets
+    DX = np.arange(0.0,-1.0,-1.0/n)
+    DX -= DX.mean()
+    DY = DX
+    for dy in DY:
+        for dx in DX:
+            offset = galsim.PositionD(dx,dy)
+            offset_list.append(offset)
+
+    np.random.seed(42) # for generating the same random permutation everytime
+    rand_idx = np.random.permutation(len(offset_list))
+    im_list_randperm = [im_list[idx] for idx in rand_idx]
+    offset_list_randperm = [offset_list[idx] for idx in rand_idx]
+    # Input to N as a tuple
+    img_randperm = galsim.utilities.interleaveImages(im_list_randperm,(n,n),offsets=offset_list_randperm)
+
+    np.testing.assert_array_equal(img_randperm.array,img.array,\
+        err_msg="Interleaved images do not match when 'offsets' is supplied")
+    assert img_randperm.scale == img.scale
+
+    # 1c) Catching errors if offsets
+    offset_list = []
+    im_list = []
+    n = 5
+    DX = np.array([-0.67,-0.33,0.,0.33,0.67])
+    DY = DX
+    for dy in DY:
+        for dx in DX:
+           offset = galsim.PositionD(dx,dy)
+           offset_list.append(offset)
+           im = galsim.Image(16,16)
+           gal.drawImage(image=im,offset=offset,scale=1.0,method='no_pixel')
+           im_list.append(im)
+
+    VE = ValueError('No error')
+    try:
+        N = (n,n)
+        galsim.utilities.interleaveImages(im_list,N=N,offsets=offset_list,catch_offset_errors=True)
+    except ValueError as VE:
+        pass
+    message =  "'offsets' must be a list of galsim.PositionD instances with x values"\
+                          +" spaced by 1/{0} and y values by 1/{1} around 0 for N = ".format(n,n)+str(N)
+    assert VE.message == message
+
+    offset_list = []
+    im_list = []
+    n = 5
+    DX = np.arange(0.,1.,1./n)
+    DY = DX
+    for dy in DY:
+        for dx in DX:
+            offset = galsim.PositionD(dx,dy)
+            offset_list.append(offset)
+            im = galsim.Image(16,16)
+            gal.drawImage(image=im,offset=offset,scale=1.0,method='no_pixel')
+            im_list.append(im)
+
+    VE = ValueError('No error')
+    try:
+        N = (n,n)
+        galsim.utilities.interleaveImages(im_list,N=N,offsets=offset_list,catch_offset_errors=True)
+    except ValueError as VE:
+        pass
+    assert VE.message == message
+
+    # 2a) Increase resolution along one direction - square to rectangular images
+    n = 2
+    g = galsim.Gaussian(sigma=3.7,flux=100.)
+    g1 = g.shear(g=1.*(n**2-1)/(n**2+1),beta=0.0*galsim.radians)
+    gal1 = g1 #galsim.Convolve([g1,galsim.Pixel(1.0)])
```

Comment:
Leaving these comments here is a bit confusing.  Either add an explicit comment that you are now interleaving surface brightness images rather than the normal pixel-convolved images that one would normally make.  Or do the normal pixel convolution here (by letting GalSim do it for you as I described above) and then later convolving by `Box(scale, scale/n/n)` and rendering with `wcs=galsim.Jacobian(scale, 0, 0, scale/n/n)`.  (At least I think that's what should work...)